### ISSUE TITLE
Interpolate before crunching

### ISSUE THREAD

**Issue by znicholls** [znicholls/silicone#18]
**Is your feature request related to a problem? Please describe.**

IAM model data often comes on different time grids. Crunching when this is the case is complicated and Silicone currently has no good, consistent solution for it. 

**Describe the solution you'd like**

Add a convenience function to interpolate data onto the same time grid before starting, which will make life easier for the crunchers.

**Describe alternatives you've considered**

Each cruncher returns 'fillers' with an `interpolate` argument, which allows the data to be interpolated as needed before being filled (and could then be returned back on the grid of the native data). I think this would be nice to have, but discussion with @Rlamboll make clear that the increase in test complexity is not ideal, especially on short timelines. This alternative could perhaps be a longer term goal.

**Additional context**

**Comment by Rlamboll** [comment]
Currently this is performed internally by the infill_all_required function. If desired I can extract what we do there and make it a separate function. Note that in this case, we infill everything (even if we don't need it for interpolating) because that function returns everything that goes into it and is supposed to be very consistent - is this still desired?

**Comment by znicholls** [comment]
I think `infill_all_required` is great but should be left where it is because it's for a very specific purpose. I think the idea of being able to interpolate as part of infilling (without everything else that happens in `infill_all_required`) is a separate one but the tests will require some thought. Hence I'd leave this open for now if that's ok with you.

**Comment by Rlamboll** [comment]
I think what might be more useful than me writing a wrapper function that does a for-loop over the (very time-consuming) interpolate function is me improving pyam interpolate to take lists in the first place, then we can just use that directly.

**Comment by znicholls** [comment]
Nice
